lib/spack/spack/package_base.py

```diff
@@ -66,6 +66,7 @@
     install_test_root,
 )
 from spack.installer import InstallError, PackageInstaller
+from spack.solver.asp import concretization_version_order
```

Comment:
Huh, what am I missing here?
```
  File "/__w/spack/spack/lib/spack/spack/package_base.py", line 69, in <module>
    from spack.solver.asp import concretization_version_order
  File "/__w/spack/spack/lib/spack/spack/solver/asp.py", line 35, in <module>
    import spack.environment as ev
AttributeError: module 'spack' has no attribute 'environment'
```

Reply:
circular imports ;p package_base depends on everything and vice versa. maybe you could move it in package_base.py

Reply:
Do we have a canonical hierarchical ordering of modules that is documented somewhere, or is this handled sort of in an ad hoc way in python?

Reply:
Moving that into `package_base.py` led to a cascade. I wasn't sure if black and flake8 would allow me to import in the function scope, but they seem fine with it. Are we concerned about performance with this approach?